### PR TITLE
Updated the Fluidic Compressor's buttons to the new Widget system.

### PR DESCRIPTION
**Pull request by iTitus**
http://imgur.com/a/WBKWz

I did not have any hovering textures, sorry for that.

### CODE REVIEW

**Comment by AEnterprise** [comment]
maybe make them list the current mode? the current implementation looks a bit confusing

File: src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java [hunk 1]
```diff
@@ -85,8 +67,45 @@ public String getInventoryName() {
...
+	}
+
+	@Override
 	public void initialize() {
-		addWidget(new WidgetFluidTank(0, (width - xSize) / 2 + 53, (height - ySize) / 2 + 16, 16, 52, this, fluidicCompressor.tank));
+		addWidget(new WidgetFluidTank(0, guiLeft + 53, guiTop + 16, 16, 52, this, fluidicCompressor.tank));
+		addWidget(new WidgetButtonFluidicCompressor(1, guiLeft + 20, guiTop + 34, 19, 16, this, fluidicCompressor.fill));
+	}
+
+	public static class WidgetButtonFluidicCompressor extends WidgetBase {
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: i prefer if you move this to a seperate file in the widgets package

File: src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java [hunk 2]
```diff
@@ -32,30 +36,8 @@ public GuiFluidicCompressor(InventoryPlayer inventoryPlayer, TileFluidicCompress
...
-	@Override
 	public void drawBackgroundPreWidgets(float f, int x, int y) {
-		if (fluidicCompressor.fill) {
-			drawTexturedModalRect(guiLeft + 20, guiTop + 25, 195, 83, 19, 16);
-			drawTexturedModalRect(guiLeft + 20, guiTop + 45, 176, 99, 19, 16);
-		} else {
-			drawTexturedModalRect(guiLeft + 20, guiTop + 45, 195, 99, 19, 16);
-			drawTexturedModalRect(guiLeft + 20, guiTop + 25, 176, 83, 19, 16);
-		}
+		drawTexturedModalRect(guiLeft + 75, guiTop + 58 - (fluidicCompressor.fill ? 39 : 0), 195, 99 - (fluidicCompressor.fill ? 16 : 0), 25, 9);
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: the widget should draw this

Comment (iTitus) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: That would be very specific for a widget. In fact it does the same as the progress bar.

File: src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java (hunk 1 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: Ok, will do

**Comment by iTitus** [comment]
The layout is from TE's Fluid Transposer button

**Comment by iTitus** [comment]
Should the tooltip show the current mode or should there be a line of text?

File: src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java [hunk 3]
```diff
@@ -85,8 +68,51 @@ public String getInventoryName() {
...
+				if (fill)
+					RenderUtils.drawImage(fillMode, x, y, width, height);
+				else
+					RenderUtils.drawImage(emptyMode, x, y, width, height);
+			}
+		}
+
+		@Override
+		public void onWidgetClicked(int x, int y, int button) {
+			fill = !fill;
```
Comment (BigXplosion) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: this logic might work but is not really logical, it should only swap when you press the light gray button

File: src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java [hunk 4]
```diff
@@ -85,8 +68,51 @@ public String getInventoryName() {
...
+		private boolean fill;
+
+		public WidgetButtonFluidicCompressor(int id, int x, int y, int width, int height, GuiBase gui, boolean fill) {
+			super(id, x, y, width, height, gui);
+			this.fill = fill;
+		}
+
+		@Override
+		public void render(int mouseX, int mouseY) {
+			if (enabled) {
```
Comment (BigXplosion) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: you don't have to check for enabled as this isn't used in this particular widget

File: src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java [hunk 5]
```diff
@@ -85,8 +68,51 @@ public String getInventoryName() {
...
+	@Override
 	public void initialize() {
-		addWidget(new WidgetFluidTank(0, (width - xSize) / 2 + 53, (height - ySize) / 2 + 16, 16, 52, this, fluidicCompressor.tank));
+		addWidget(new WidgetFluidTank(0, guiLeft + 53, guiTop + 16, 16, 52, this, fluidicCompressor.tank));
+		addWidget(new WidgetButtonFluidicCompressor(1, guiLeft + 20, guiTop + 34, 19, 16, this, fluidicCompressor.fill));
+	}
+
+	public static class WidgetButtonFluidicCompressor extends WidgetBase {
+
+		private static final ResourceLocation fillMode = new ResourceLocation("bcadditions", "textures/gui/Pieces/fillMode.png");
```
Comment (BigXplosion) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: Don't put those widgets in different files... there is no need for it as these icons are only used for this machine

File: src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java [hunk 6]
```diff
@@ -85,8 +68,51 @@ public String getInventoryName() {
...
+		private static final ResourceLocation emptyMode = new ResourceLocation("bcadditions", "textures/gui/Pieces/emptyMode.png");
```
Comment (BigXplosion) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: Same as above

File: src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java (hunk 1 above)
Comment (BigXplosion) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: There is a WidgetButtonUpdate that would perfectly fit the needs for this? why re-invent the wheel

File: src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java [hunk 7]
```diff
@@ -85,8 +68,51 @@ public String getInventoryName() {
 	}
 
 	@Override
+	public void widgetActionPerformed(WidgetBase widget) {
+		if (widget instanceof WidgetButtonFluidicCompressor) {
+			fluidicCompressor.fill = ((WidgetButtonFluidicCompressor) widget).fill;
+			PacketHandler.instance.sendToServer(new MessageFluidicCompressorA(fluidicCompressor.fill, fluidicCompressor));
```
Comment (BigXplosion) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: You can use a MessageWidgetUpdate for this, and remove this packet

Comment (iTitus) [inline src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java]: Sorry, I missunderstood AEnterprises comment. 
I dont know how the other messages work, so i used the one that is already there. I did not refractor the GUI Code :P
Will look into it later this evening.

**Comment by BigXplosion** [comment]
@iTitus, don't reference to TE but look at other code in BCA that does the same thing.

See this: https://github.com/AEnterprise/Buildcraft-Additions/blob/multiblockRefinery/src/main/java/buildcraftAdditions/client/gui/gui/GuiPipeColoringTool.java#L112
this: https://github.com/AEnterprise/Buildcraft-Additions/blob/multiblockRefinery/src/main/java/buildcraftAdditions/client/gui/gui/GuiPipeColoringTool.java#L94
and this: https://github.com/AEnterprise/Buildcraft-Additions/blob/multiblockRefinery/src/main/java/buildcraftAdditions/client/gui/widgets/WidgetButtonUpdate.java

**Pull request closed without merging**